Modules/Forum/classes/class.ilObjForumGUI.php

```diff
@@ -3078,22 +3097,49 @@ public function viewThreadObject()
                     }
                 }
                 $this->renderPostContent($threadContentTemplate, $node, $this->requestAction, $pageIndex, $postIndex);
-                $this->renderDraftContent($threadContentTemplate, $this->requestAction, $render_drafts, $node, $selected_draft_id);
+                if ( ilForumPostDraft::isSavePostDraftAllowed() && $_SESSION['viewmode'] < ilForumProperties::VIEW_DATE) {
```

Comment:
Why do you use the `<` operator here? I'd expect something like `===` or `!=='.

Reply:
No. Because the "$currentSortation === ilForumProperties::VIEW_TREE" condition does not match in every case when it is expected. 
I am still not sure why this happens, but the $currentSortation variable isn't even set properly in every changed viewmode case. 

The tests showed, that the session var is set properly instead.